// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/model/backgroundImage.js
"use strict";
/**
 * Blog Post endpoints
 * \"Use these endpoints for interacting with Blog Posts, Blog Authors, and Blog Tags\"
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackgroundImage = void 0;
class BackgroundImage {
    static getAttributeTypeMap() {
        return BackgroundImage.attributeTypeMap;
    }
}
exports.BackgroundImage = BackgroundImage;
BackgroundImage.discriminator = undefined;
BackgroundImage.attributeTypeMap = [
    {
        "name": "imageUrl",
        "baseName": "imageUrl",
        "type": "string"
    },
    {
        "name": "backgroundSize",
        "baseName": "backgroundSize",
        "type": "string"
    },
    {
        "name": "backgroundPosition",
        "baseName": "backgroundPosition",
        "type": "BackgroundImage.BackgroundPositionEnum"
    }
];
(function (BackgroundImage) {
    let BackgroundPositionEnum;
    (function (BackgroundPositionEnum) {
        BackgroundPositionEnum[BackgroundPositionEnum["TopLeft"] = 'TOP_LEFT'] = "TopLeft";
        BackgroundPositionEnum[BackgroundPositionEnum["TopCenter"] = 'TOP_CENTER'] = "TopCenter";
        BackgroundPositionEnum[BackgroundPositionEnum["TopRight"] = 'TOP_RIGHT'] = "TopRight";
        BackgroundPositionEnum[BackgroundPositionEnum["MiddleLeft"] = 'MIDDLE_LEFT'] = "MiddleLeft";
        BackgroundPositionEnum[BackgroundPositionEnum["MiddleCenter"] = 'MIDDLE_CENTER'] = "MiddleCenter";
        BackgroundPositionEnum[BackgroundPositionEnum["MiddleRight"] = 'MIDDLE_RIGHT'] = "MiddleRight";
        BackgroundPositionEnum[BackgroundPositionEnum["BottomLeft"] = 'BOTTOM_LEFT'] = "BottomLeft";
        BackgroundPositionEnum[BackgroundPositionEnum["BottomCenter"] = 'BOTTOM_CENTER'] = "BottomCenter";
        BackgroundPositionEnum[BackgroundPositionEnum["BottomRight"] = 'BOTTOM_RIGHT'] = "BottomRight";
    })(BackgroundPositionEnum = BackgroundImage.BackgroundPositionEnum || (BackgroundImage.BackgroundPositionEnum = {}));
})(BackgroundImage = exports.BackgroundImage || (exports.BackgroundImage = {}));
