// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/model/sideOrCorner.js
"use strict";
/**
 * Blog Post endpoints
 * \"Use these endpoints for interacting with Blog Posts, Blog Authors, and Blog Tags\"
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SideOrCorner = void 0;
class SideOrCorner {
    static getAttributeTypeMap() {
        return SideOrCorner.attributeTypeMap;
    }
}
exports.SideOrCorner = SideOrCorner;
SideOrCorner.discriminator = undefined;
SideOrCorner.attributeTypeMap = [
    {
        "name": "verticalSide",
        "baseName": "verticalSide",
        "type": "SideOrCorner.VerticalSideEnum"
    },
    {
        "name": "horizontalSide",
        "baseName": "horizontalSide",
        "type": "SideOrCorner.HorizontalSideEnum"
    }
];
(function (SideOrCorner) {
    let VerticalSideEnum;
    (function (VerticalSideEnum) {
        VerticalSideEnum[VerticalSideEnum["Top"] = 'TOP'] = "Top";
        VerticalSideEnum[VerticalSideEnum["Middle"] = 'MIDDLE'] = "Middle";
        VerticalSideEnum[VerticalSideEnum["Bottom"] = 'BOTTOM'] = "Bottom";
    })(VerticalSideEnum = SideOrCorner.VerticalSideEnum || (SideOrCorner.VerticalSideEnum = {}));
    let HorizontalSideEnum;
    (function (HorizontalSideEnum) {
        HorizontalSideEnum[HorizontalSideEnum["Left"] = 'LEFT'] = "Left";
        HorizontalSideEnum[HorizontalSideEnum["Center"] = 'CENTER'] = "Center";
        HorizontalSideEnum[HorizontalSideEnum["Right"] = 'RIGHT'] = "Right";
    })(HorizontalSideEnum = SideOrCorner.HorizontalSideEnum || (SideOrCorner.HorizontalSideEnum = {}));
})(SideOrCorner = exports.SideOrCorner || (exports.SideOrCorner = {}));
