// @hubspot/api-client/lib/codegen/cms/blogs/blog_posts/model/styles.js
"use strict";
/**
 * Blog Post endpoints
 * \"Use these endpoints for interacting with Blog Posts, Blog Authors, and Blog Tags\"
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Styles = void 0;
class Styles {
    static getAttributeTypeMap() {
        return Styles.attributeTypeMap;
    }
}
exports.Styles = Styles;
Styles.discriminator = undefined;
Styles.attributeTypeMap = [
    {
        "name": "verticalAlignment",
        "baseName": "verticalAlignment",
        "type": "Styles.VerticalAlignmentEnum"
    },
    {
        "name": "backgroundColor",
        "baseName": "backgroundColor",
        "type": "RGBAColor"
    },
    {
        "name": "backgroundImage",
        "baseName": "backgroundImage",
        "type": "BackgroundImage"
    },
    {
        "name": "backgroundGradient",
        "baseName": "backgroundGradient",
        "type": "Gradient"
    },
    {
        "name": "maxWidthSectionCentering",
        "baseName": "maxWidthSectionCentering",
        "type": "number"
    },
    {
        "name": "forceFullWidthSection",
        "baseName": "forceFullWidthSection",
        "type": "boolean"
    },
    {
        "name": "flexboxPositioning",
        "baseName": "flexboxPositioning",
        "type": "Styles.FlexboxPositioningEnum"
    }
];
(function (Styles) {
    let VerticalAlignmentEnum;
    (function (VerticalAlignmentEnum) {
        VerticalAlignmentEnum[VerticalAlignmentEnum["Top"] = 'TOP'] = "Top";
        VerticalAlignmentEnum[VerticalAlignmentEnum["Middle"] = 'MIDDLE'] = "Middle";
        VerticalAlignmentEnum[VerticalAlignmentEnum["Bottom"] = 'BOTTOM'] = "Bottom";
    })(VerticalAlignmentEnum = Styles.VerticalAlignmentEnum || (Styles.VerticalAlignmentEnum = {}));
    let FlexboxPositioningEnum;
    (function (FlexboxPositioningEnum) {
        FlexboxPositioningEnum[FlexboxPositioningEnum["TopLeft"] = 'TOP_LEFT'] = "TopLeft";
        FlexboxPositioningEnum[FlexboxPositioningEnum["TopCenter"] = 'TOP_CENTER'] = "TopCenter";
        FlexboxPositioningEnum[FlexboxPositioningEnum["TopRight"] = 'TOP_RIGHT'] = "TopRight";
        FlexboxPositioningEnum[FlexboxPositioningEnum["MiddleLeft"] = 'MIDDLE_LEFT'] = "MiddleLeft";
        FlexboxPositioningEnum[FlexboxPositioningEnum["MiddleCenter"] = 'MIDDLE_CENTER'] = "MiddleCenter";
        FlexboxPositioningEnum[FlexboxPositioningEnum["MiddleRight"] = 'MIDDLE_RIGHT'] = "MiddleRight";
        FlexboxPositioningEnum[FlexboxPositioningEnum["BottomLeft"] = 'BOTTOM_LEFT'] = "BottomLeft";
        FlexboxPositioningEnum[FlexboxPositioningEnum["BottomCenter"] = 'BOTTOM_CENTER'] = "BottomCenter";
        FlexboxPositioningEnum[FlexboxPositioningEnum["BottomRight"] = 'BOTTOM_RIGHT'] = "BottomRight";
    })(FlexboxPositioningEnum = Styles.FlexboxPositioningEnum || (Styles.FlexboxPositioningEnum = {}));
})(Styles = exports.Styles || (exports.Styles = {}));
