// @hubspot/api-client/lib/codegen/cms/performance/model/publicPerformanceResponse.js
"use strict";
/**
 * CMS Performance API
 * Use these endpoints to get a time series view of your website\'s performance.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicPerformanceResponse = void 0;
class PublicPerformanceResponse {
    static getAttributeTypeMap() {
        return PublicPerformanceResponse.attributeTypeMap;
    }
}
exports.PublicPerformanceResponse = PublicPerformanceResponse;
PublicPerformanceResponse.discriminator = undefined;
PublicPerformanceResponse.attributeTypeMap = [
    {
        "name": "data",
        "baseName": "data",
        "type": "Array<PerformanceView>"
    },
    {
        "name": "domain",
        "baseName": "domain",
        "type": "string"
    },
    {
        "name": "path",
        "baseName": "path",
        "type": "string"
    },
    {
        "name": "startInterval",
        "baseName": "startInterval",
        "type": "number"
    },
    {
        "name": "endInterval",
        "baseName": "endInterval",
        "type": "number"
    },
    {
        "name": "interval",
        "baseName": "interval",
        "type": "PublicPerformanceResponse.IntervalEnum"
    },
    {
        "name": "period",
        "baseName": "period",
        "type": "PublicPerformanceResponse.PeriodEnum"
    }
];
(function (PublicPerformanceResponse) {
    let IntervalEnum;
    (function (IntervalEnum) {
        IntervalEnum[IntervalEnum["OneMinute"] = 'ONE_MINUTE'] = "OneMinute";
        IntervalEnum[IntervalEnum["FiveMinutes"] = 'FIVE_MINUTES'] = "FiveMinutes";
        IntervalEnum[IntervalEnum["TenMinutes"] = 'TEN_MINUTES'] = "TenMinutes";
        IntervalEnum[IntervalEnum["FifteenMinutes"] = 'FIFTEEN_MINUTES'] = "FifteenMinutes";
        IntervalEnum[IntervalEnum["ThirtyMinutes"] = 'THIRTY_MINUTES'] = "ThirtyMinutes";
        IntervalEnum[IntervalEnum["OneHour"] = 'ONE_HOUR'] = "OneHour";
        IntervalEnum[IntervalEnum["FourHours"] = 'FOUR_HOURS'] = "FourHours";
        IntervalEnum[IntervalEnum["TwelveHours"] = 'TWELVE_HOURS'] = "TwelveHours";
        IntervalEnum[IntervalEnum["OneDay"] = 'ONE_DAY'] = "OneDay";
        IntervalEnum[IntervalEnum["OneWeek"] = 'ONE_WEEK'] = "OneWeek";
    })(IntervalEnum = PublicPerformanceResponse.IntervalEnum || (PublicPerformanceResponse.IntervalEnum = {}));
    let PeriodEnum;
    (function (PeriodEnum) {
        PeriodEnum[PeriodEnum["OneMinute"] = 'ONE_MINUTE'] = "OneMinute";
        PeriodEnum[PeriodEnum["FiveMinutes"] = 'FIVE_MINUTES'] = "FiveMinutes";
        PeriodEnum[PeriodEnum["TenMinutes"] = 'TEN_MINUTES'] = "TenMinutes";
        PeriodEnum[PeriodEnum["FifteenMinutes"] = 'FIFTEEN_MINUTES'] = "FifteenMinutes";
        PeriodEnum[PeriodEnum["ThirtyMinutes"] = 'THIRTY_MINUTES'] = "ThirtyMinutes";
        PeriodEnum[PeriodEnum["OneHour"] = 'ONE_HOUR'] = "OneHour";
        PeriodEnum[PeriodEnum["FourHours"] = 'FOUR_HOURS'] = "FourHours";
        PeriodEnum[PeriodEnum["TwelveHours"] = 'TWELVE_HOURS'] = "TwelveHours";
        PeriodEnum[PeriodEnum["OneDay"] = 'ONE_DAY'] = "OneDay";
        PeriodEnum[PeriodEnum["OneWeek"] = 'ONE_WEEK'] = "OneWeek";
    })(PeriodEnum = PublicPerformanceResponse.PeriodEnum || (PublicPerformanceResponse.PeriodEnum = {}));
})(PublicPerformanceResponse = exports.PublicPerformanceResponse || (exports.PublicPerformanceResponse = {}));
