// @hubspot/api-client/lib/codegen/cms/domains/model/domain.js
"use strict";
/**
 * Domains
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Domain = void 0;
class Domain {
    static getAttributeTypeMap() {
        return Domain.attributeTypeMap;
    }
}
exports.Domain = Domain;
Domain.discriminator = undefined;
Domain.attributeTypeMap = [
    {
        "name": "portalId",
        "baseName": "portalId",
        "type": "number"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "number"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "number"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "number"
    },
    {
        "name": "domain",
        "baseName": "domain",
        "type": "string"
    },
    {
        "name": "primaryLandingPage",
        "baseName": "primaryLandingPage",
        "type": "boolean"
    },
    {
        "name": "primaryEmail",
        "baseName": "primaryEmail",
        "type": "boolean"
    },
    {
        "name": "primaryBlog",
        "baseName": "primaryBlog",
        "type": "boolean"
    },
    {
        "name": "primaryBlogPost",
        "baseName": "primaryBlogPost",
        "type": "boolean"
    },
    {
        "name": "primarySitePage",
        "baseName": "primarySitePage",
        "type": "boolean"
    },
    {
        "name": "primaryKnowledge",
        "baseName": "primaryKnowledge",
        "type": "boolean"
    },
    {
        "name": "primaryLegacyPage",
        "baseName": "primaryLegacyPage",
        "type": "boolean"
    },
    {
        "name": "primaryClickTracking",
        "baseName": "primaryClickTracking",
        "type": "boolean"
    },
    {
        "name": "fullCategoryKey",
        "baseName": "fullCategoryKey",
        "type": "string"
    },
    {
        "name": "secondaryToDomain",
        "baseName": "secondaryToDomain",
        "type": "string"
    },
    {
        "name": "isResolving",
        "baseName": "isResolving",
        "type": "boolean"
    },
    {
        "name": "isDnsCorrect",
        "baseName": "isDnsCorrect",
        "type": "boolean"
    },
    {
        "name": "manuallyMarkedAsResolving",
        "baseName": "manuallyMarkedAsResolving",
        "type": "boolean"
    },
    {
        "name": "consecutiveNonResolvingCount",
        "baseName": "consecutiveNonResolvingCount",
        "type": "number"
    },
    {
        "name": "sslCname",
        "baseName": "sslCname",
        "type": "string"
    },
    {
        "name": "isSslEnabled",
        "baseName": "isSslEnabled",
        "type": "boolean"
    },
    {
        "name": "isSslOnly",
        "baseName": "isSslOnly",
        "type": "boolean"
    },
    {
        "name": "certificateId",
        "baseName": "certificateId",
        "type": "number"
    },
    {
        "name": "sslRequestId",
        "baseName": "sslRequestId",
        "type": "number"
    },
    {
        "name": "isUsedForBlogPost",
        "baseName": "isUsedForBlogPost",
        "type": "boolean"
    },
    {
        "name": "isUsedForSitePage",
        "baseName": "isUsedForSitePage",
        "type": "boolean"
    },
    {
        "name": "isUsedForLandingPage",
        "baseName": "isUsedForLandingPage",
        "type": "boolean"
    },
    {
        "name": "isUsedForEmail",
        "baseName": "isUsedForEmail",
        "type": "boolean"
    },
    {
        "name": "isUsedForKnowledge",
        "baseName": "isUsedForKnowledge",
        "type": "boolean"
    },
    {
        "name": "setupTaskId",
        "baseName": "setupTaskId",
        "type": "number"
    },
    {
        "name": "isSetupComplete",
        "baseName": "isSetupComplete",
        "type": "boolean"
    },
    {
        "name": "setUpLanguage",
        "baseName": "setUpLanguage",
        "type": "string"
    },
    {
        "name": "teamIds",
        "baseName": "teamIds",
        "type": "Array<number>"
    },
    {
        "name": "actualCname",
        "baseName": "actualCname",
        "type": "string"
    },
    {
        "name": "correctCname",
        "baseName": "correctCname",
        "type": "string"
    },
    {
        "name": "actualIp",
        "baseName": "actualIp",
        "type": "string"
    },
    {
        "name": "apexResolutionStatus",
        "baseName": "apexResolutionStatus",
        "type": "Domain.ApexResolutionStatusEnum"
    },
    {
        "name": "apexDomain",
        "baseName": "apexDomain",
        "type": "string"
    },
    {
        "name": "publicSuffix",
        "baseName": "publicSuffix",
        "type": "string"
    },
    {
        "name": "apexIpAddresses",
        "baseName": "apexIpAddresses",
        "type": "Array<string>"
    },
    {
        "name": "siteId",
        "baseName": "siteId",
        "type": "number"
    },
    {
        "name": "brandId",
        "baseName": "brandId",
        "type": "number"
    },
    {
        "name": "deletable",
        "baseName": "deletable",
        "type": "boolean"
    },
    {
        "name": "domainCdnConfig",
        "baseName": "domainCdnConfig",
        "type": "DomainCdnConfig"
    },
    {
        "name": "setupInfo",
        "baseName": "setupInfo",
        "type": "DomainSetupInfo"
    },
    {
        "name": "derivedBrandName",
        "baseName": "derivedBrandName",
        "type": "string"
    },
    {
        "name": "createdById",
        "baseName": "createdById",
        "type": "number"
    },
    {
        "name": "updatedById",
        "baseName": "updatedById",
        "type": "number"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "isAnyPrimary",
        "baseName": "isAnyPrimary",
        "type": "boolean"
    },
    {
        "name": "isLegacyDomain",
        "baseName": "isLegacyDomain",
        "type": "boolean"
    },
    {
        "name": "isInternalDomain",
        "baseName": "isInternalDomain",
        "type": "boolean"
    },
    {
        "name": "isResolvingInternalProperty",
        "baseName": "isResolvingInternalProperty",
        "type": "boolean"
    },
    {
        "name": "isResolvingIgnoringManuallyMarkedAsResolving",
        "baseName": "isResolvingIgnoringManuallyMarkedAsResolving",
        "type": "boolean"
    },
    {
        "name": "isUsedForAnyContentType",
        "baseName": "isUsedForAnyContentType",
        "type": "boolean"
    },
    {
        "name": "isLegacy",
        "baseName": "isLegacy",
        "type": "boolean"
    },
    {
        "name": "authorAt",
        "baseName": "authorAt",
        "type": "number"
    },
    {
        "name": "cosObjectType",
        "baseName": "cosObjectType",
        "type": "Domain.CosObjectTypeEnum"
    },
    {
        "name": "cdnPurgeEmbargoTime",
        "baseName": "cdnPurgeEmbargoTime",
        "type": "number"
    },
    {
        "name": "isStagingDomain",
        "baseName": "isStagingDomain",
        "type": "boolean"
    }
];
(function (Domain) {
    let ApexResolutionStatusEnum;
    (function (ApexResolutionStatusEnum) {
        ApexResolutionStatusEnum[ApexResolutionStatusEnum["Ineligible"] = 'INELIGIBLE'] = "Ineligible";
        ApexResolutionStatusEnum[ApexResolutionStatusEnum["SuggestResolving"] = 'SUGGEST_RESOLVING'] = "SuggestResolving";
        ApexResolutionStatusEnum[ApexResolutionStatusEnum["AlreadyResolving"] = 'ALREADY_RESOLVING'] = "AlreadyResolving";
        ApexResolutionStatusEnum[ApexResolutionStatusEnum["Error"] = 'ERROR'] = "Error";
    })(ApexResolutionStatusEnum = Domain.ApexResolutionStatusEnum || (Domain.ApexResolutionStatusEnum = {}));
    let CosObjectTypeEnum;
    (function (CosObjectTypeEnum) {
        CosObjectTypeEnum[CosObjectTypeEnum["Content"] = 'CONTENT'] = "Content";
        CosObjectTypeEnum[CosObjectTypeEnum["ExtensionResource"] = 'EXTENSION_RESOURCE'] = "ExtensionResource";
        CosObjectTypeEnum[CosObjectTypeEnum["Layout"] = 'LAYOUT'] = "Layout";
        CosObjectTypeEnum[CosObjectTypeEnum["CustomWidget"] = 'CUSTOM_WIDGET'] = "CustomWidget";
        CosObjectTypeEnum[CosObjectTypeEnum["Widget"] = 'WIDGET'] = "Widget";
        CosObjectTypeEnum[CosObjectTypeEnum["Form"] = 'FORM'] = "Form";
        CosObjectTypeEnum[CosObjectTypeEnum["Placement"] = 'PLACEMENT'] = "Placement";
        CosObjectTypeEnum[CosObjectTypeEnum["Image"] = 'IMAGE'] = "Image";
        CosObjectTypeEnum[CosObjectTypeEnum["DomainSettings"] = 'DOMAIN_SETTINGS'] = "DomainSettings";
        CosObjectTypeEnum[CosObjectTypeEnum["SiteSettings"] = 'SITE_SETTINGS'] = "SiteSettings";
        CosObjectTypeEnum[CosObjectTypeEnum["EmailAddress"] = 'EMAIL_ADDRESS'] = "EmailAddress";
        CosObjectTypeEnum[CosObjectTypeEnum["Workflow"] = 'WORKFLOW'] = "Workflow";
        CosObjectTypeEnum[CosObjectTypeEnum["HubdbTable"] = 'HUBDB_TABLE'] = "HubdbTable";
        CosObjectTypeEnum[CosObjectTypeEnum["RedirectUrl"] = 'REDIRECT_URL'] = "RedirectUrl";
        CosObjectTypeEnum[CosObjectTypeEnum["DesignFolder"] = 'DESIGN_FOLDER'] = "DesignFolder";
        CosObjectTypeEnum[CosObjectTypeEnum["SiteMap"] = 'SITE_MAP'] = "SiteMap";
        CosObjectTypeEnum[CosObjectTypeEnum["Domain"] = 'DOMAIN'] = "Domain";
        CosObjectTypeEnum[CosObjectTypeEnum["Blog"] = 'BLOG'] = "Blog";
        CosObjectTypeEnum[CosObjectTypeEnum["File"] = 'FILE'] = "File";
        CosObjectTypeEnum[CosObjectTypeEnum["Folder"] = 'FOLDER'] = "Folder";
        CosObjectTypeEnum[CosObjectTypeEnum["SiteMenu"] = 'SITE_MENU'] = "SiteMenu";
        CosObjectTypeEnum[CosObjectTypeEnum["Theme"] = 'THEME'] = "Theme";
        CosObjectTypeEnum[CosObjectTypeEnum["ContentGroup"] = 'CONTENT_GROUP'] = "ContentGroup";
        CosObjectTypeEnum[CosObjectTypeEnum["FollowMe"] = 'FOLLOW_ME'] = "FollowMe";
        CosObjectTypeEnum[CosObjectTypeEnum["KnowledgeBase"] = 'KNOWLEDGE_BASE'] = "KnowledgeBase";
        CosObjectTypeEnum[CosObjectTypeEnum["ListMembership"] = 'LIST_MEMBERSHIP'] = "ListMembership";
        CosObjectTypeEnum[CosObjectTypeEnum["ContactMembership"] = 'CONTACT_MEMBERSHIP'] = "ContactMembership";
        CosObjectTypeEnum[CosObjectTypeEnum["PasswordProtected"] = 'PASSWORD_PROTECTED'] = "PasswordProtected";
        CosObjectTypeEnum[CosObjectTypeEnum["UnrestrictedAccess"] = 'UNRESTRICTED_ACCESS'] = "UnrestrictedAccess";
        CosObjectTypeEnum[CosObjectTypeEnum["MarketplaceListing"] = 'MARKETPLACE_LISTING'] = "MarketplaceListing";
        CosObjectTypeEnum[CosObjectTypeEnum["LayoutSection"] = 'LAYOUT_SECTION'] = "LayoutSection";
        CosObjectTypeEnum[CosObjectTypeEnum["ThemeSettings"] = 'THEME_SETTINGS'] = "ThemeSettings";
        CosObjectTypeEnum[CosObjectTypeEnum["VideoPlayer"] = 'VIDEO_PLAYER'] = "VideoPlayer";
        CosObjectTypeEnum[CosObjectTypeEnum["UrlMapping"] = 'URL_MAPPING'] = "UrlMapping";
        CosObjectTypeEnum[CosObjectTypeEnum["KnowledgeCategory"] = 'KNOWLEDGE_CATEGORY'] = "KnowledgeCategory";
        CosObjectTypeEnum[CosObjectTypeEnum["KnowledgeHomepageCategory"] = 'KNOWLEDGE_HOMEPAGE_CATEGORY'] = "KnowledgeHomepageCategory";
        CosObjectTypeEnum[CosObjectTypeEnum["RawAsset"] = 'RAW_ASSET'] = "RawAsset";
        CosObjectTypeEnum[CosObjectTypeEnum["GlobalContent"] = 'GLOBAL_CONTENT'] = "GlobalContent";
        CosObjectTypeEnum[CosObjectTypeEnum["HubdbTableRow"] = 'HUBDB_TABLE_ROW'] = "HubdbTableRow";
        CosObjectTypeEnum[CosObjectTypeEnum["BlogAuthor"] = 'BLOG_AUTHOR'] = "BlogAuthor";
        CosObjectTypeEnum[CosObjectTypeEnum["ServerlessFunction"] = 'SERVERLESS_FUNCTION'] = "ServerlessFunction";
        CosObjectTypeEnum[CosObjectTypeEnum["KnowledgeCategoryTranslation"] = 'KNOWLEDGE_CATEGORY_TRANSLATION'] = "KnowledgeCategoryTranslation";
    })(CosObjectTypeEnum = Domain.CosObjectTypeEnum || (Domain.CosObjectTypeEnum = {}));
})(Domain = exports.Domain || (exports.Domain = {}));
