// @hubspot/api-client/lib/codegen/files/model/fileUpdateInput.js
"use strict";
/**
 * Files
 * Upload and manage files.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileUpdateInput = void 0;
/**
* Object for updating files.
*/
class FileUpdateInput {
    static getAttributeTypeMap() {
        return FileUpdateInput.attributeTypeMap;
    }
}
exports.FileUpdateInput = FileUpdateInput;
FileUpdateInput.discriminator = undefined;
FileUpdateInput.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "parentFolderId",
        "baseName": "parentFolderId",
        "type": "string"
    },
    {
        "name": "parentFolderPath",
        "baseName": "parentFolderPath",
        "type": "string"
    },
    {
        "name": "isUsableInContent",
        "baseName": "isUsableInContent",
        "type": "boolean"
    },
    {
        "name": "access",
        "baseName": "access",
        "type": "FileUpdateInput.AccessEnum"
    }
];
(function (FileUpdateInput) {
    let AccessEnum;
    (function (AccessEnum) {
        AccessEnum[AccessEnum["PublicIndexable"] = 'PUBLIC_INDEXABLE'] = "PublicIndexable";
        AccessEnum[AccessEnum["PublicNotIndexable"] = 'PUBLIC_NOT_INDEXABLE'] = "PublicNotIndexable";
        AccessEnum[AccessEnum["HiddenIndexable"] = 'HIDDEN_INDEXABLE'] = "HiddenIndexable";
        AccessEnum[AccessEnum["HiddenNotIndexable"] = 'HIDDEN_NOT_INDEXABLE'] = "HiddenNotIndexable";
        AccessEnum[AccessEnum["HiddenPrivate"] = 'HIDDEN_PRIVATE'] = "HiddenPrivate";
        AccessEnum[AccessEnum["Private"] = 'PRIVATE'] = "Private";
    })(AccessEnum = FileUpdateInput.AccessEnum || (FileUpdateInput.AccessEnum = {}));
})(FileUpdateInput = exports.FileUpdateInput || (exports.FileUpdateInput = {}));
