// @hubspot/api-client/lib/codegen/cms/audit_logs/model/publicAuditLog.js
"use strict";
/**
 * CMS Audit Logs
 * Use this endpoint to query audit logs of CMS changes that occurred on your HubSpot account.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicAuditLog = void 0;
class PublicAuditLog {
    static getAttributeTypeMap() {
        return PublicAuditLog.attributeTypeMap;
    }
}
exports.PublicAuditLog = PublicAuditLog;
PublicAuditLog.discriminator = undefined;
PublicAuditLog.attributeTypeMap = [
    {
        "name": "objectId",
        "baseName": "objectId",
        "type": "string"
    },
    {
        "name": "userId",
        "baseName": "userId",
        "type": "string"
    },
    {
        "name": "timestamp",
        "baseName": "timestamp",
        "type": "Date"
    },
    {
        "name": "objectName",
        "baseName": "objectName",
        "type": "string"
    },
    {
        "name": "fullName",
        "baseName": "fullName",
        "type": "string"
    },
    {
        "name": "event",
        "baseName": "event",
        "type": "PublicAuditLog.EventEnum"
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "PublicAuditLog.ObjectTypeEnum"
    }
];
(function (PublicAuditLog) {
    let EventEnum;
    (function (EventEnum) {
        EventEnum[EventEnum["Created"] = 'CREATED'] = "Created";
        EventEnum[EventEnum["Updated"] = 'UPDATED'] = "Updated";
        EventEnum[EventEnum["Published"] = 'PUBLISHED'] = "Published";
        EventEnum[EventEnum["Deleted"] = 'DELETED'] = "Deleted";
        EventEnum[EventEnum["Unpublished"] = 'UNPUBLISHED'] = "Unpublished";
    })(EventEnum = PublicAuditLog.EventEnum || (PublicAuditLog.EventEnum = {}));
    let ObjectTypeEnum;
    (function (ObjectTypeEnum) {
        ObjectTypeEnum[ObjectTypeEnum["Blog"] = 'BLOG'] = "Blog";
        ObjectTypeEnum[ObjectTypeEnum["BlogPost"] = 'BLOG_POST'] = "BlogPost";
        ObjectTypeEnum[ObjectTypeEnum["LandingPage"] = 'LANDING_PAGE'] = "LandingPage";
        ObjectTypeEnum[ObjectTypeEnum["WebsitePage"] = 'WEBSITE_PAGE'] = "WebsitePage";
        ObjectTypeEnum[ObjectTypeEnum["Template"] = 'TEMPLATE'] = "Template";
        ObjectTypeEnum[ObjectTypeEnum["Module"] = 'MODULE'] = "Module";
        ObjectTypeEnum[ObjectTypeEnum["GlobalModule"] = 'GLOBAL_MODULE'] = "GlobalModule";
        ObjectTypeEnum[ObjectTypeEnum["ServerlessFunction"] = 'SERVERLESS_FUNCTION'] = "ServerlessFunction";
        ObjectTypeEnum[ObjectTypeEnum["Domain"] = 'DOMAIN'] = "Domain";
        ObjectTypeEnum[ObjectTypeEnum["UrlMapping"] = 'URL_MAPPING'] = "UrlMapping";
        ObjectTypeEnum[ObjectTypeEnum["Email"] = 'EMAIL'] = "Email";
        ObjectTypeEnum[ObjectTypeEnum["ContentSettings"] = 'CONTENT_SETTINGS'] = "ContentSettings";
        ObjectTypeEnum[ObjectTypeEnum["HubdbTable"] = 'HUBDB_TABLE'] = "HubdbTable";
        ObjectTypeEnum[ObjectTypeEnum["KnowledgeBaseArticle"] = 'KNOWLEDGE_BASE_ARTICLE'] = "KnowledgeBaseArticle";
        ObjectTypeEnum[ObjectTypeEnum["KnowledgeBase"] = 'KNOWLEDGE_BASE'] = "KnowledgeBase";
        ObjectTypeEnum[ObjectTypeEnum["Theme"] = 'THEME'] = "Theme";
        ObjectTypeEnum[ObjectTypeEnum["Css"] = 'CSS'] = "Css";
        ObjectTypeEnum[ObjectTypeEnum["Js"] = 'JS'] = "Js";
    })(ObjectTypeEnum = PublicAuditLog.ObjectTypeEnum || (PublicAuditLog.ObjectTypeEnum = {}));
})(PublicAuditLog = exports.PublicAuditLog || (exports.PublicAuditLog = {}));
