// @hubspot/api-client/lib/codegen/crm/associations/model/batchResponsePublicAssociation.js
"use strict";
/**
 * Associations
 * Associations define the relationships between objects in HubSpot. These endpoints allow you to create, read, and remove associations.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponsePublicAssociation = void 0;
class BatchResponsePublicAssociation {
    static getAttributeTypeMap() {
        return BatchResponsePublicAssociation.attributeTypeMap;
    }
}
exports.BatchResponsePublicAssociation = BatchResponsePublicAssociation;
BatchResponsePublicAssociation.discriminator = undefined;
BatchResponsePublicAssociation.attributeTypeMap = [
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<PublicAssociation>"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number"
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponsePublicAssociation.StatusEnum"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    }
];
(function (BatchResponsePublicAssociation) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Pending"] = 'PENDING'] = "Pending";
        StatusEnum[StatusEnum["Processing"] = 'PROCESSING'] = "Processing";
        StatusEnum[StatusEnum["Canceled"] = 'CANCELED'] = "Canceled";
        StatusEnum[StatusEnum["Complete"] = 'COMPLETE'] = "Complete";
    })(StatusEnum = BatchResponsePublicAssociation.StatusEnum || (BatchResponsePublicAssociation.StatusEnum = {}));
})(BatchResponsePublicAssociation = exports.BatchResponsePublicAssociation || (exports.BatchResponsePublicAssociation = {}));
