// @hubspot/api-client/lib/codegen/communication_preferences/model/publicSubscriptionStatus.js
"use strict";
/**
 * Subscriptions
 * Subscriptions allow contacts to control what forms of communications they receive. Contacts can decide whether they want to receive communication pertaining to a specific topic, brand, or an entire HubSpot account.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicSubscriptionStatus = void 0;
/**
* The status of a subscription for a contact.
*/
class PublicSubscriptionStatus {
    static getAttributeTypeMap() {
        return PublicSubscriptionStatus.attributeTypeMap;
    }
}
exports.PublicSubscriptionStatus = PublicSubscriptionStatus;
PublicSubscriptionStatus.discriminator = undefined;
PublicSubscriptionStatus.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "description",
        "baseName": "description",
        "type": "string"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "PublicSubscriptionStatus.StatusEnum"
    },
    {
        "name": "sourceOfStatus",
        "baseName": "sourceOfStatus",
        "type": "PublicSubscriptionStatus.SourceOfStatusEnum"
    },
    {
        "name": "brandId",
        "baseName": "brandId",
        "type": "number"
    },
    {
        "name": "preferenceGroupName",
        "baseName": "preferenceGroupName",
        "type": "string"
    },
    {
        "name": "legalBasis",
        "baseName": "legalBasis",
        "type": "PublicSubscriptionStatus.LegalBasisEnum"
    },
    {
        "name": "legalBasisExplanation",
        "baseName": "legalBasisExplanation",
        "type": "string"
    }
];
(function (PublicSubscriptionStatus) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Subscribed"] = 'SUBSCRIBED'] = "Subscribed";
        StatusEnum[StatusEnum["NotSubscribed"] = 'NOT_SUBSCRIBED'] = "NotSubscribed";
    })(StatusEnum = PublicSubscriptionStatus.StatusEnum || (PublicSubscriptionStatus.StatusEnum = {}));
    let SourceOfStatusEnum;
    (function (SourceOfStatusEnum) {
        SourceOfStatusEnum[SourceOfStatusEnum["PortalWideStatus"] = 'PORTAL_WIDE_STATUS'] = "PortalWideStatus";
        SourceOfStatusEnum[SourceOfStatusEnum["BrandWideStatus"] = 'BRAND_WIDE_STATUS'] = "BrandWideStatus";
        SourceOfStatusEnum[SourceOfStatusEnum["SubscriptionStatus"] = 'SUBSCRIPTION_STATUS'] = "SubscriptionStatus";
    })(SourceOfStatusEnum = PublicSubscriptionStatus.SourceOfStatusEnum || (PublicSubscriptionStatus.SourceOfStatusEnum = {}));
    let LegalBasisEnum;
    (function (LegalBasisEnum) {
        LegalBasisEnum[LegalBasisEnum["LegitimateInterestPql"] = 'LEGITIMATE_INTEREST_PQL'] = "LegitimateInterestPql";
        LegalBasisEnum[LegalBasisEnum["LegitimateInterestClient"] = 'LEGITIMATE_INTEREST_CLIENT'] = "LegitimateInterestClient";
        LegalBasisEnum[LegalBasisEnum["PerformanceOfContract"] = 'PERFORMANCE_OF_CONTRACT'] = "PerformanceOfContract";
        LegalBasisEnum[LegalBasisEnum["ConsentWithNotice"] = 'CONSENT_WITH_NOTICE'] = "ConsentWithNotice";
        LegalBasisEnum[LegalBasisEnum["NonGdpr"] = 'NON_GDPR'] = "NonGdpr";
        LegalBasisEnum[LegalBasisEnum["ProcessAndStore"] = 'PROCESS_AND_STORE'] = "ProcessAndStore";
        LegalBasisEnum[LegalBasisEnum["LegitimateInterestOther"] = 'LEGITIMATE_INTEREST_OTHER'] = "LegitimateInterestOther";
    })(LegalBasisEnum = PublicSubscriptionStatus.LegalBasisEnum || (PublicSubscriptionStatus.LegalBasisEnum = {}));
})(PublicSubscriptionStatus = exports.PublicSubscriptionStatus || (exports.PublicSubscriptionStatus = {}));
