// @hubspot/api-client/lib/codegen/marketing/transactional/model/emailSendStatusView.js
"use strict";
/**
 * Transactional Email
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailSendStatusView = void 0;
/**
* Describes the status of an email send request.
*/
class EmailSendStatusView {
    static getAttributeTypeMap() {
        return EmailSendStatusView.attributeTypeMap;
    }
}
exports.EmailSendStatusView = EmailSendStatusView;
EmailSendStatusView.discriminator = undefined;
EmailSendStatusView.attributeTypeMap = [
    {
        "name": "eventId",
        "baseName": "eventId",
        "type": "EventIdView"
    },
    {
        "name": "statusId",
        "baseName": "statusId",
        "type": "string"
    },
    {
        "name": "sendResult",
        "baseName": "sendResult",
        "type": "EmailSendStatusView.SendResultEnum"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "status",
        "baseName": "status",
        "type": "EmailSendStatusView.StatusEnum"
    }
];
(function (EmailSendStatusView) {
    let SendResultEnum;
    (function (SendResultEnum) {
        SendResultEnum[SendResultEnum["Sent"] = 'SENT'] = "Sent";
        SendResultEnum[SendResultEnum["IdempotentIgnore"] = 'IDEMPOTENT_IGNORE'] = "IdempotentIgnore";
        SendResultEnum[SendResultEnum["Queued"] = 'QUEUED'] = "Queued";
        SendResultEnum[SendResultEnum["IdempotentFail"] = 'IDEMPOTENT_FAIL'] = "IdempotentFail";
        SendResultEnum[SendResultEnum["Throttled"] = 'THROTTLED'] = "Throttled";
        SendResultEnum[SendResultEnum["EmailDisabled"] = 'EMAIL_DISABLED'] = "EmailDisabled";
        SendResultEnum[SendResultEnum["PortalSuspended"] = 'PORTAL_SUSPENDED'] = "PortalSuspended";
        SendResultEnum[SendResultEnum["InvalidToAddress"] = 'INVALID_TO_ADDRESS'] = "InvalidToAddress";
        SendResultEnum[SendResultEnum["BlockedDomain"] = 'BLOCKED_DOMAIN'] = "BlockedDomain";
        SendResultEnum[SendResultEnum["PreviouslyBounced"] = 'PREVIOUSLY_BOUNCED'] = "PreviouslyBounced";
        SendResultEnum[SendResultEnum["EmailUnconfirmed"] = 'EMAIL_UNCONFIRMED'] = "EmailUnconfirmed";
        SendResultEnum[SendResultEnum["PreviousSpam"] = 'PREVIOUS_SPAM'] = "PreviousSpam";
        SendResultEnum[SendResultEnum["PreviouslyUnsubscribedMessage"] = 'PREVIOUSLY_UNSUBSCRIBED_MESSAGE'] = "PreviouslyUnsubscribedMessage";
        SendResultEnum[SendResultEnum["PreviouslyUnsubscribedPortal"] = 'PREVIOUSLY_UNSUBSCRIBED_PORTAL'] = "PreviouslyUnsubscribedPortal";
        SendResultEnum[SendResultEnum["InvalidFromAddress"] = 'INVALID_FROM_ADDRESS'] = "InvalidFromAddress";
        SendResultEnum[SendResultEnum["CampaignCancelled"] = 'CAMPAIGN_CANCELLED'] = "CampaignCancelled";
        SendResultEnum[SendResultEnum["ValidationFailed"] = 'VALIDATION_FAILED'] = "ValidationFailed";
        SendResultEnum[SendResultEnum["MtaIgnore"] = 'MTA_IGNORE'] = "MtaIgnore";
        SendResultEnum[SendResultEnum["BlockedAddress"] = 'BLOCKED_ADDRESS'] = "BlockedAddress";
        SendResultEnum[SendResultEnum["PortalOverLimit"] = 'PORTAL_OVER_LIMIT'] = "PortalOverLimit";
        SendResultEnum[SendResultEnum["PortalExpired"] = 'PORTAL_EXPIRED'] = "PortalExpired";
        SendResultEnum[SendResultEnum["PortalMissingMarketingScope"] = 'PORTAL_MISSING_MARKETING_SCOPE'] = "PortalMissingMarketingScope";
        SendResultEnum[SendResultEnum["MissingTemplateProperties"] = 'MISSING_TEMPLATE_PROPERTIES'] = "MissingTemplateProperties";
        SendResultEnum[SendResultEnum["MissingRequiredParameter"] = 'MISSING_REQUIRED_PARAMETER'] = "MissingRequiredParameter";
        SendResultEnum[SendResultEnum["PortalAuthenticationFailure"] = 'PORTAL_AUTHENTICATION_FAILURE'] = "PortalAuthenticationFailure";
        SendResultEnum[SendResultEnum["MissingContent"] = 'MISSING_CONTENT'] = "MissingContent";
        SendResultEnum[SendResultEnum["CorruptInput"] = 'CORRUPT_INPUT'] = "CorruptInput";
        SendResultEnum[SendResultEnum["TemplateRenderException"] = 'TEMPLATE_RENDER_EXCEPTION'] = "TemplateRenderException";
        SendResultEnum[SendResultEnum["GraymailSuppressed"] = 'GRAYMAIL_SUPPRESSED'] = "GraymailSuppressed";
        SendResultEnum[SendResultEnum["UnconfiguredSendingDomain"] = 'UNCONFIGURED_SENDING_DOMAIN'] = "UnconfiguredSendingDomain";
        SendResultEnum[SendResultEnum["Undeliverable"] = 'UNDELIVERABLE'] = "Undeliverable";
        SendResultEnum[SendResultEnum["CancelledAbuse"] = 'CANCELLED_ABUSE'] = "CancelledAbuse";
        SendResultEnum[SendResultEnum["QuarantinedAddress"] = 'QUARANTINED_ADDRESS'] = "QuarantinedAddress";
        SendResultEnum[SendResultEnum["AddressOnlyAcceptedOnProd"] = 'ADDRESS_ONLY_ACCEPTED_ON_PROD'] = "AddressOnlyAcceptedOnProd";
        SendResultEnum[SendResultEnum["PortalNotAuthorizedForApplication"] = 'PORTAL_NOT_AUTHORIZED_FOR_APPLICATION'] = "PortalNotAuthorizedForApplication";
        SendResultEnum[SendResultEnum["AddressListBombed"] = 'ADDRESS_LIST_BOMBED'] = "AddressListBombed";
        SendResultEnum[SendResultEnum["AddressOptedOut"] = 'ADDRESS_OPTED_OUT'] = "AddressOptedOut";
        SendResultEnum[SendResultEnum["RecipientFatigueSuppressed"] = 'RECIPIENT_FATIGUE_SUPPRESSED'] = "RecipientFatigueSuppressed";
        SendResultEnum[SendResultEnum["TooManyRecipients"] = 'TOO_MANY_RECIPIENTS'] = "TooManyRecipients";
        SendResultEnum[SendResultEnum["PreviouslyUnsubscribedBrand"] = 'PREVIOUSLY_UNSUBSCRIBED_BRAND'] = "PreviouslyUnsubscribedBrand";
        SendResultEnum[SendResultEnum["NonMarketableContact"] = 'NON_MARKETABLE_CONTACT'] = "NonMarketableContact";
    })(SendResultEnum = EmailSendStatusView.SendResultEnum || (EmailSendStatusView.SendResultEnum = {}));
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Pending"] = 'PENDING'] = "Pending";
        StatusEnum[StatusEnum["Processing"] = 'PROCESSING'] = "Processing";
        StatusEnum[StatusEnum["Canceled"] = 'CANCELED'] = "Canceled";
        StatusEnum[StatusEnum["Complete"] = 'COMPLETE'] = "Complete";
    })(StatusEnum = EmailSendStatusView.StatusEnum || (EmailSendStatusView.StatusEnum = {}));
})(EmailSendStatusView = exports.EmailSendStatusView || (exports.EmailSendStatusView = {}));
