// @hubspot/api-client/lib/codegen/cms/url_redirects/model/urlMapping.js
"use strict";
/**
 * URL redirects
 * URL redirect operations
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UrlMapping = void 0;
class UrlMapping {
    static getAttributeTypeMap() {
        return UrlMapping.attributeTypeMap;
    }
}
exports.UrlMapping = UrlMapping;
UrlMapping.discriminator = undefined;
UrlMapping.attributeTypeMap = [
    {
        "name": "id",
        "baseName": "id",
        "type": "number"
    },
    {
        "name": "portalId",
        "baseName": "portalId",
        "type": "number"
    },
    {
        "name": "created",
        "baseName": "created",
        "type": "number"
    },
    {
        "name": "updated",
        "baseName": "updated",
        "type": "number"
    },
    {
        "name": "createdById",
        "baseName": "createdById",
        "type": "number"
    },
    {
        "name": "updatedById",
        "baseName": "updatedById",
        "type": "number"
    },
    {
        "name": "routePrefix",
        "baseName": "routePrefix",
        "type": "string"
    },
    {
        "name": "destination",
        "baseName": "destination",
        "type": "string"
    },
    {
        "name": "redirectStyle",
        "baseName": "redirectStyle",
        "type": "number"
    },
    {
        "name": "contentGroupId",
        "baseName": "contentGroupId",
        "type": "number"
    },
    {
        "name": "isOnlyAfterNotFound",
        "baseName": "isOnlyAfterNotFound",
        "type": "boolean"
    },
    {
        "name": "isRegex",
        "baseName": "isRegex",
        "type": "boolean"
    },
    {
        "name": "isMatchFullUrl",
        "baseName": "isMatchFullUrl",
        "type": "boolean"
    },
    {
        "name": "isMatchQueryString",
        "baseName": "isMatchQueryString",
        "type": "boolean"
    },
    {
        "name": "isPattern",
        "baseName": "isPattern",
        "type": "boolean"
    },
    {
        "name": "isTrailingSlashOptional",
        "baseName": "isTrailingSlashOptional",
        "type": "boolean"
    },
    {
        "name": "isProtocolAgnostic",
        "baseName": "isProtocolAgnostic",
        "type": "boolean"
    },
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "precedence",
        "baseName": "precedence",
        "type": "number"
    },
    {
        "name": "deletedAt",
        "baseName": "deletedAt",
        "type": "number"
    },
    {
        "name": "note",
        "baseName": "note",
        "type": "string"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "internallyCreated",
        "baseName": "internallyCreated",
        "type": "boolean"
    },
    {
        "name": "cosObjectType",
        "baseName": "cosObjectType",
        "type": "UrlMapping.CosObjectTypeEnum"
    },
    {
        "name": "cdnPurgeEmbargoTime",
        "baseName": "cdnPurgeEmbargoTime",
        "type": "number"
    }
];
(function (UrlMapping) {
    let CosObjectTypeEnum;
    (function (CosObjectTypeEnum) {
        CosObjectTypeEnum[CosObjectTypeEnum["Content"] = 'CONTENT'] = "Content";
        CosObjectTypeEnum[CosObjectTypeEnum["ExtensionResource"] = 'EXTENSION_RESOURCE'] = "ExtensionResource";
        CosObjectTypeEnum[CosObjectTypeEnum["Layout"] = 'LAYOUT'] = "Layout";
        CosObjectTypeEnum[CosObjectTypeEnum["CustomWidget"] = 'CUSTOM_WIDGET'] = "CustomWidget";
        CosObjectTypeEnum[CosObjectTypeEnum["Widget"] = 'WIDGET'] = "Widget";
        CosObjectTypeEnum[CosObjectTypeEnum["Form"] = 'FORM'] = "Form";
        CosObjectTypeEnum[CosObjectTypeEnum["Placement"] = 'PLACEMENT'] = "Placement";
        CosObjectTypeEnum[CosObjectTypeEnum["Image"] = 'IMAGE'] = "Image";
        CosObjectTypeEnum[CosObjectTypeEnum["DomainSettings"] = 'DOMAIN_SETTINGS'] = "DomainSettings";
        CosObjectTypeEnum[CosObjectTypeEnum["SiteSettings"] = 'SITE_SETTINGS'] = "SiteSettings";
        CosObjectTypeEnum[CosObjectTypeEnum["EmailAddress"] = 'EMAIL_ADDRESS'] = "EmailAddress";
        CosObjectTypeEnum[CosObjectTypeEnum["Workflow"] = 'WORKFLOW'] = "Workflow";
        CosObjectTypeEnum[CosObjectTypeEnum["HubdbTable"] = 'HUBDB_TABLE'] = "HubdbTable";
        CosObjectTypeEnum[CosObjectTypeEnum["RedirectUrl"] = 'REDIRECT_URL'] = "RedirectUrl";
        CosObjectTypeEnum[CosObjectTypeEnum["DesignFolder"] = 'DESIGN_FOLDER'] = "DesignFolder";
        CosObjectTypeEnum[CosObjectTypeEnum["SiteMap"] = 'SITE_MAP'] = "SiteMap";
        CosObjectTypeEnum[CosObjectTypeEnum["Domain"] = 'DOMAIN'] = "Domain";
        CosObjectTypeEnum[CosObjectTypeEnum["Blog"] = 'BLOG'] = "Blog";
        CosObjectTypeEnum[CosObjectTypeEnum["File"] = 'FILE'] = "File";
        CosObjectTypeEnum[CosObjectTypeEnum["Folder"] = 'FOLDER'] = "Folder";
        CosObjectTypeEnum[CosObjectTypeEnum["SiteMenu"] = 'SITE_MENU'] = "SiteMenu";
        CosObjectTypeEnum[CosObjectTypeEnum["Theme"] = 'THEME'] = "Theme";
        CosObjectTypeEnum[CosObjectTypeEnum["ContentGroup"] = 'CONTENT_GROUP'] = "ContentGroup";
        CosObjectTypeEnum[CosObjectTypeEnum["FollowMe"] = 'FOLLOW_ME'] = "FollowMe";
        CosObjectTypeEnum[CosObjectTypeEnum["KnowledgeBase"] = 'KNOWLEDGE_BASE'] = "KnowledgeBase";
        CosObjectTypeEnum[CosObjectTypeEnum["ListMembership"] = 'LIST_MEMBERSHIP'] = "ListMembership";
        CosObjectTypeEnum[CosObjectTypeEnum["ContactMembership"] = 'CONTACT_MEMBERSHIP'] = "ContactMembership";
        CosObjectTypeEnum[CosObjectTypeEnum["PasswordProtected"] = 'PASSWORD_PROTECTED'] = "PasswordProtected";
        CosObjectTypeEnum[CosObjectTypeEnum["UnrestrictedAccess"] = 'UNRESTRICTED_ACCESS'] = "UnrestrictedAccess";
        CosObjectTypeEnum[CosObjectTypeEnum["MarketplaceListing"] = 'MARKETPLACE_LISTING'] = "MarketplaceListing";
        CosObjectTypeEnum[CosObjectTypeEnum["LayoutSection"] = 'LAYOUT_SECTION'] = "LayoutSection";
        CosObjectTypeEnum[CosObjectTypeEnum["ThemeSettings"] = 'THEME_SETTINGS'] = "ThemeSettings";
        CosObjectTypeEnum[CosObjectTypeEnum["VideoPlayer"] = 'VIDEO_PLAYER'] = "VideoPlayer";
        CosObjectTypeEnum[CosObjectTypeEnum["UrlMapping"] = 'URL_MAPPING'] = "UrlMapping";
        CosObjectTypeEnum[CosObjectTypeEnum["KnowledgeCategory"] = 'KNOWLEDGE_CATEGORY'] = "KnowledgeCategory";
        CosObjectTypeEnum[CosObjectTypeEnum["KnowledgeHomepageCategory"] = 'KNOWLEDGE_HOMEPAGE_CATEGORY'] = "KnowledgeHomepageCategory";
        CosObjectTypeEnum[CosObjectTypeEnum["RawAsset"] = 'RAW_ASSET'] = "RawAsset";
        CosObjectTypeEnum[CosObjectTypeEnum["GlobalContent"] = 'GLOBAL_CONTENT'] = "GlobalContent";
        CosObjectTypeEnum[CosObjectTypeEnum["HubdbTableRow"] = 'HUBDB_TABLE_ROW'] = "HubdbTableRow";
        CosObjectTypeEnum[CosObjectTypeEnum["BlogAuthor"] = 'BLOG_AUTHOR'] = "BlogAuthor";
        CosObjectTypeEnum[CosObjectTypeEnum["ServerlessFunction"] = 'SERVERLESS_FUNCTION'] = "ServerlessFunction";
        CosObjectTypeEnum[CosObjectTypeEnum["KnowledgeCategoryTranslation"] = 'KNOWLEDGE_CATEGORY_TRANSLATION'] = "KnowledgeCategoryTranslation";
    })(CosObjectTypeEnum = UrlMapping.CosObjectTypeEnum || (UrlMapping.CosObjectTypeEnum = {}));
})(UrlMapping = exports.UrlMapping || (exports.UrlMapping = {}));
