// @hubspot/api-client/lib/codegen/crm/imports/model/actionResponse.js
"use strict";
/**
 * CRM Imports
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionResponse = void 0;
class ActionResponse {
    static getAttributeTypeMap() {
        return ActionResponse.attributeTypeMap;
    }
}
exports.ActionResponse = ActionResponse;
ActionResponse.discriminator = undefined;
ActionResponse.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "ActionResponse.StatusEnum"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    }
];
(function (ActionResponse) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Pending"] = 'PENDING'] = "Pending";
        StatusEnum[StatusEnum["Processing"] = 'PROCESSING'] = "Processing";
        StatusEnum[StatusEnum["Canceled"] = 'CANCELED'] = "Canceled";
        StatusEnum[StatusEnum["Complete"] = 'COMPLETE'] = "Complete";
    })(StatusEnum = ActionResponse.StatusEnum || (ActionResponse.StatusEnum = {}));
})(ActionResponse = exports.ActionResponse || (exports.ActionResponse = {}));
