// @hubspot/api-client/lib/codegen/crm/imports/model/publicImportError.js
"use strict";
/**
 * CRM Imports
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PublicImportError = void 0;
class PublicImportError {
    static getAttributeTypeMap() {
        return PublicImportError.attributeTypeMap;
    }
}
exports.PublicImportError = PublicImportError;
PublicImportError.discriminator = undefined;
PublicImportError.attributeTypeMap = [
    {
        "name": "errorType",
        "baseName": "errorType",
        "type": "PublicImportError.ErrorTypeEnum"
    },
    {
        "name": "objectType",
        "baseName": "objectType",
        "type": "PublicImportError.ObjectTypeEnum"
    },
    {
        "name": "invalidValue",
        "baseName": "invalidValue",
        "type": "string"
    },
    {
        "name": "extraContext",
        "baseName": "extraContext",
        "type": "string"
    },
    {
        "name": "objectTypeId",
        "baseName": "objectTypeId",
        "type": "string"
    },
    {
        "name": "knownColumnNumber",
        "baseName": "knownColumnNumber",
        "type": "number"
    },
    {
        "name": "sourceData",
        "baseName": "sourceData",
        "type": "ImportRowCore"
    },
    {
        "name": "createdAt",
        "baseName": "createdAt",
        "type": "number"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
(function (PublicImportError) {
    let ErrorTypeEnum;
    (function (ErrorTypeEnum) {
        ErrorTypeEnum[ErrorTypeEnum["IncorrectNumberOfColumns"] = 'INCORRECT_NUMBER_OF_COLUMNS'] = "IncorrectNumberOfColumns";
        ErrorTypeEnum[ErrorTypeEnum["InvalidObjectId"] = 'INVALID_OBJECT_ID'] = "InvalidObjectId";
        ErrorTypeEnum[ErrorTypeEnum["InvalidAssociationIdentifier"] = 'INVALID_ASSOCIATION_IDENTIFIER'] = "InvalidAssociationIdentifier";
        ErrorTypeEnum[ErrorTypeEnum["NoObjectIdFromAssociationIdentifier"] = 'NO_OBJECT_ID_FROM_ASSOCIATION_IDENTIFIER'] = "NoObjectIdFromAssociationIdentifier";
        ErrorTypeEnum[ErrorTypeEnum["MultipleCompaniesWithThisDomain"] = 'MULTIPLE_COMPANIES_WITH_THIS_DOMAIN'] = "MultipleCompaniesWithThisDomain";
        ErrorTypeEnum[ErrorTypeEnum["PropertyDefinitionNotFound"] = 'PROPERTY_DEFINITION_NOT_FOUND'] = "PropertyDefinitionNotFound";
        ErrorTypeEnum[ErrorTypeEnum["PropertyValueNotFound"] = 'PROPERTY_VALUE_NOT_FOUND'] = "PropertyValueNotFound";
        ErrorTypeEnum[ErrorTypeEnum["CouldNotFindOwner"] = 'COULD_NOT_FIND_OWNER'] = "CouldNotFindOwner";
        ErrorTypeEnum[ErrorTypeEnum["MultipleOwnersFound"] = 'MULTIPLE_OWNERS_FOUND'] = "MultipleOwnersFound";
        ErrorTypeEnum[ErrorTypeEnum["CouldNotParseNumber"] = 'COULD_NOT_PARSE_NUMBER'] = "CouldNotParseNumber";
        ErrorTypeEnum[ErrorTypeEnum["CouldNotParseDate"] = 'COULD_NOT_PARSE_DATE'] = "CouldNotParseDate";
        ErrorTypeEnum[ErrorTypeEnum["CouldNotParseTerm"] = 'COULD_NOT_PARSE_TERM'] = "CouldNotParseTerm";
        ErrorTypeEnum[ErrorTypeEnum["OutsideValidTimeRange"] = 'OUTSIDE_VALID_TIME_RANGE'] = "OutsideValidTimeRange";
        ErrorTypeEnum[ErrorTypeEnum["OutsideValidTermRange"] = 'OUTSIDE_VALID_TERM_RANGE'] = "OutsideValidTermRange";
        ErrorTypeEnum[ErrorTypeEnum["CouldNotParseRow"] = 'COULD_NOT_PARSE_ROW'] = "CouldNotParseRow";
        ErrorTypeEnum[ErrorTypeEnum["InvalidEnumerationOption"] = 'INVALID_ENUMERATION_OPTION'] = "InvalidEnumerationOption";
        ErrorTypeEnum[ErrorTypeEnum["AmbiguousEnumerationOption"] = 'AMBIGUOUS_ENUMERATION_OPTION'] = "AmbiguousEnumerationOption";
        ErrorTypeEnum[ErrorTypeEnum["FailedValidation"] = 'FAILED_VALIDATION'] = "FailedValidation";
        ErrorTypeEnum[ErrorTypeEnum["FailedToCreateAssociation"] = 'FAILED_TO_CREATE_ASSOCIATION'] = "FailedToCreateAssociation";
        ErrorTypeEnum[ErrorTypeEnum["FileNotFound"] = 'FILE_NOT_FOUND'] = "FileNotFound";
        ErrorTypeEnum[ErrorTypeEnum["InvalidColumnConfiguration"] = 'INVALID_COLUMN_CONFIGURATION'] = "InvalidColumnConfiguration";
        ErrorTypeEnum[ErrorTypeEnum["InvalidFileType"] = 'INVALID_FILE_TYPE'] = "InvalidFileType";
        ErrorTypeEnum[ErrorTypeEnum["InvalidSpreadsheet"] = 'INVALID_SPREADSHEET'] = "InvalidSpreadsheet";
        ErrorTypeEnum[ErrorTypeEnum["InvalidSheetCount"] = 'INVALID_SHEET_COUNT'] = "InvalidSheetCount";
        ErrorTypeEnum[ErrorTypeEnum["FailedToProcessObjectWithEmptyPropertyValues"] = 'FAILED_TO_PROCESS_OBJECT_WITH_EMPTY_PROPERTY_VALUES'] = "FailedToProcessObjectWithEmptyPropertyValues";
        ErrorTypeEnum[ErrorTypeEnum["UnknownBadRequest"] = 'UNKNOWN_BAD_REQUEST'] = "UnknownBadRequest";
        ErrorTypeEnum[ErrorTypeEnum["GdprBlacklistedEmail"] = 'GDPR_BLACKLISTED_EMAIL'] = "GdprBlacklistedEmail";
        ErrorTypeEnum[ErrorTypeEnum["DuplicateAssociationId"] = 'DUPLICATE_ASSOCIATION_ID'] = "DuplicateAssociationId";
        ErrorTypeEnum[ErrorTypeEnum["LimitExceeded"] = 'LIMIT_EXCEEDED'] = "LimitExceeded";
        ErrorTypeEnum[ErrorTypeEnum["InvalidAlternateId"] = 'INVALID_ALTERNATE_ID'] = "InvalidAlternateId";
        ErrorTypeEnum[ErrorTypeEnum["InvalidEmail"] = 'INVALID_EMAIL'] = "InvalidEmail";
        ErrorTypeEnum[ErrorTypeEnum["InvalidDomain"] = 'INVALID_DOMAIN'] = "InvalidDomain";
        ErrorTypeEnum[ErrorTypeEnum["DuplicateRowContent"] = 'DUPLICATE_ROW_CONTENT'] = "DuplicateRowContent";
        ErrorTypeEnum[ErrorTypeEnum["InvalidNumberSize"] = 'INVALID_NUMBER_SIZE'] = "InvalidNumberSize";
        ErrorTypeEnum[ErrorTypeEnum["UnknownError"] = 'UNKNOWN_ERROR'] = "UnknownError";
        ErrorTypeEnum[ErrorTypeEnum["FailedToOptOutContact"] = 'FAILED_TO_OPT_OUT_CONTACT'] = "FailedToOptOutContact";
        ErrorTypeEnum[ErrorTypeEnum["InvalidRequiredProperty"] = 'INVALID_REQUIRED_PROPERTY'] = "InvalidRequiredProperty";
        ErrorTypeEnum[ErrorTypeEnum["DuplicateAlternateId"] = 'DUPLICATE_ALTERNATE_ID'] = "DuplicateAlternateId";
        ErrorTypeEnum[ErrorTypeEnum["DuplicateObjectId"] = 'DUPLICATE_OBJECT_ID'] = "DuplicateObjectId";
        ErrorTypeEnum[ErrorTypeEnum["DuplicateUniquePropertyValue"] = 'DUPLICATE_UNIQUE_PROPERTY_VALUE'] = "DuplicateUniquePropertyValue";
    })(ErrorTypeEnum = PublicImportError.ErrorTypeEnum || (PublicImportError.ErrorTypeEnum = {}));
    let ObjectTypeEnum;
    (function (ObjectTypeEnum) {
        ObjectTypeEnum[ObjectTypeEnum["Contact"] = 'CONTACT'] = "Contact";
        ObjectTypeEnum[ObjectTypeEnum["Company"] = 'COMPANY'] = "Company";
        ObjectTypeEnum[ObjectTypeEnum["Deal"] = 'DEAL'] = "Deal";
        ObjectTypeEnum[ObjectTypeEnum["Engagement"] = 'ENGAGEMENT'] = "Engagement";
        ObjectTypeEnum[ObjectTypeEnum["Ticket"] = 'TICKET'] = "Ticket";
        ObjectTypeEnum[ObjectTypeEnum["Owner"] = 'OWNER'] = "Owner";
        ObjectTypeEnum[ObjectTypeEnum["Product"] = 'PRODUCT'] = "Product";
        ObjectTypeEnum[ObjectTypeEnum["LineItem"] = 'LINE_ITEM'] = "LineItem";
        ObjectTypeEnum[ObjectTypeEnum["BetDeliverableService"] = 'BET_DELIVERABLE_SERVICE'] = "BetDeliverableService";
        ObjectTypeEnum[ObjectTypeEnum["Content"] = 'CONTENT'] = "Content";
        ObjectTypeEnum[ObjectTypeEnum["Conversation"] = 'CONVERSATION'] = "Conversation";
        ObjectTypeEnum[ObjectTypeEnum["BetAlert"] = 'BET_ALERT'] = "BetAlert";
        ObjectTypeEnum[ObjectTypeEnum["Portal"] = 'PORTAL'] = "Portal";
        ObjectTypeEnum[ObjectTypeEnum["Quote"] = 'QUOTE'] = "Quote";
        ObjectTypeEnum[ObjectTypeEnum["FormSubmissionInbounddb"] = 'FORM_SUBMISSION_INBOUNDDB'] = "FormSubmissionInbounddb";
        ObjectTypeEnum[ObjectTypeEnum["Quota"] = 'QUOTA'] = "Quota";
        ObjectTypeEnum[ObjectTypeEnum["Unsubscribe"] = 'UNSUBSCRIBE'] = "Unsubscribe";
        ObjectTypeEnum[ObjectTypeEnum["Communication"] = 'COMMUNICATION'] = "Communication";
        ObjectTypeEnum[ObjectTypeEnum["FeedbackSubmission"] = 'FEEDBACK_SUBMISSION'] = "FeedbackSubmission";
        ObjectTypeEnum[ObjectTypeEnum["Attribution"] = 'ATTRIBUTION'] = "Attribution";
        ObjectTypeEnum[ObjectTypeEnum["SalesforceSyncError"] = 'SALESFORCE_SYNC_ERROR'] = "SalesforceSyncError";
        ObjectTypeEnum[ObjectTypeEnum["RestorableCrmObject"] = 'RESTORABLE_CRM_OBJECT'] = "RestorableCrmObject";
        ObjectTypeEnum[ObjectTypeEnum["Hub"] = 'HUB'] = "Hub";
        ObjectTypeEnum[ObjectTypeEnum["LandingPage"] = 'LANDING_PAGE'] = "LandingPage";
        ObjectTypeEnum[ObjectTypeEnum["ProductOrFolder"] = 'PRODUCT_OR_FOLDER'] = "ProductOrFolder";
        ObjectTypeEnum[ObjectTypeEnum["Task"] = 'TASK'] = "Task";
        ObjectTypeEnum[ObjectTypeEnum["Form"] = 'FORM'] = "Form";
        ObjectTypeEnum[ObjectTypeEnum["MarketingEmail"] = 'MARKETING_EMAIL'] = "MarketingEmail";
        ObjectTypeEnum[ObjectTypeEnum["AdAccount"] = 'AD_ACCOUNT'] = "AdAccount";
        ObjectTypeEnum[ObjectTypeEnum["AdCampaign"] = 'AD_CAMPAIGN'] = "AdCampaign";
        ObjectTypeEnum[ObjectTypeEnum["AdGroup"] = 'AD_GROUP'] = "AdGroup";
        ObjectTypeEnum[ObjectTypeEnum["Ad"] = 'AD'] = "Ad";
        ObjectTypeEnum[ObjectTypeEnum["Keyword"] = 'KEYWORD'] = "Keyword";
        ObjectTypeEnum[ObjectTypeEnum["Campaign"] = 'CAMPAIGN'] = "Campaign";
        ObjectTypeEnum[ObjectTypeEnum["SocialChannel"] = 'SOCIAL_CHANNEL'] = "SocialChannel";
        ObjectTypeEnum[ObjectTypeEnum["SocialPost"] = 'SOCIAL_POST'] = "SocialPost";
        ObjectTypeEnum[ObjectTypeEnum["SitePage"] = 'SITE_PAGE'] = "SitePage";
        ObjectTypeEnum[ObjectTypeEnum["BlogPost"] = 'BLOG_POST'] = "BlogPost";
        ObjectTypeEnum[ObjectTypeEnum["Import"] = 'IMPORT'] = "Import";
        ObjectTypeEnum[ObjectTypeEnum["Export"] = 'EXPORT'] = "Export";
        ObjectTypeEnum[ObjectTypeEnum["Cta"] = 'CTA'] = "Cta";
        ObjectTypeEnum[ObjectTypeEnum["TaskTemplate"] = 'TASK_TEMPLATE'] = "TaskTemplate";
        ObjectTypeEnum[ObjectTypeEnum["AutomationPlatformFlow"] = 'AUTOMATION_PLATFORM_FLOW'] = "AutomationPlatformFlow";
        ObjectTypeEnum[ObjectTypeEnum["ObjectList"] = 'OBJECT_LIST'] = "ObjectList";
        ObjectTypeEnum[ObjectTypeEnum["Note"] = 'NOTE'] = "Note";
        ObjectTypeEnum[ObjectTypeEnum["MeetingEvent"] = 'MEETING_EVENT'] = "MeetingEvent";
        ObjectTypeEnum[ObjectTypeEnum["Call"] = 'CALL'] = "Call";
        ObjectTypeEnum[ObjectTypeEnum["Email"] = 'EMAIL'] = "Email";
        ObjectTypeEnum[ObjectTypeEnum["PublishingTask"] = 'PUBLISHING_TASK'] = "PublishingTask";
        ObjectTypeEnum[ObjectTypeEnum["ConversationSession"] = 'CONVERSATION_SESSION'] = "ConversationSession";
        ObjectTypeEnum[ObjectTypeEnum["ContactCreateAttribution"] = 'CONTACT_CREATE_ATTRIBUTION'] = "ContactCreateAttribution";
        ObjectTypeEnum[ObjectTypeEnum["Invoice"] = 'INVOICE'] = "Invoice";
        ObjectTypeEnum[ObjectTypeEnum["MarketingEvent"] = 'MARKETING_EVENT'] = "MarketingEvent";
        ObjectTypeEnum[ObjectTypeEnum["ConversationInbox"] = 'CONVERSATION_INBOX'] = "ConversationInbox";
        ObjectTypeEnum[ObjectTypeEnum["Chatflow"] = 'CHATFLOW'] = "Chatflow";
        ObjectTypeEnum[ObjectTypeEnum["MediaBridge"] = 'MEDIA_BRIDGE'] = "MediaBridge";
        ObjectTypeEnum[ObjectTypeEnum["Sequence"] = 'SEQUENCE'] = "Sequence";
        ObjectTypeEnum[ObjectTypeEnum["SequenceStep"] = 'SEQUENCE_STEP'] = "SequenceStep";
        ObjectTypeEnum[ObjectTypeEnum["Forecast"] = 'FORECAST'] = "Forecast";
        ObjectTypeEnum[ObjectTypeEnum["Snippet"] = 'SNIPPET'] = "Snippet";
        ObjectTypeEnum[ObjectTypeEnum["Template"] = 'TEMPLATE'] = "Template";
        ObjectTypeEnum[ObjectTypeEnum["DealCreateAttribution"] = 'DEAL_CREATE_ATTRIBUTION'] = "DealCreateAttribution";
        ObjectTypeEnum[ObjectTypeEnum["QuoteTemplate"] = 'QUOTE_TEMPLATE'] = "QuoteTemplate";
        ObjectTypeEnum[ObjectTypeEnum["QuoteModule"] = 'QUOTE_MODULE'] = "QuoteModule";
        ObjectTypeEnum[ObjectTypeEnum["QuoteModuleField"] = 'QUOTE_MODULE_FIELD'] = "QuoteModuleField";
        ObjectTypeEnum[ObjectTypeEnum["QuoteField"] = 'QUOTE_FIELD'] = "QuoteField";
        ObjectTypeEnum[ObjectTypeEnum["SequenceEnrollment"] = 'SEQUENCE_ENROLLMENT'] = "SequenceEnrollment";
        ObjectTypeEnum[ObjectTypeEnum["Subscription"] = 'SUBSCRIPTION'] = "Subscription";
        ObjectTypeEnum[ObjectTypeEnum["AcceptanceTest"] = 'ACCEPTANCE_TEST'] = "AcceptanceTest";
        ObjectTypeEnum[ObjectTypeEnum["SocialBroadcast"] = 'SOCIAL_BROADCAST'] = "SocialBroadcast";
        ObjectTypeEnum[ObjectTypeEnum["DealSplit"] = 'DEAL_SPLIT'] = "DealSplit";
        ObjectTypeEnum[ObjectTypeEnum["DealRegistration"] = 'DEAL_REGISTRATION'] = "DealRegistration";
        ObjectTypeEnum[ObjectTypeEnum["GoalTarget"] = 'GOAL_TARGET'] = "GoalTarget";
        ObjectTypeEnum[ObjectTypeEnum["GoalTargetGroup"] = 'GOAL_TARGET_GROUP'] = "GoalTargetGroup";
        ObjectTypeEnum[ObjectTypeEnum["PortalObjectSyncMessage"] = 'PORTAL_OBJECT_SYNC_MESSAGE'] = "PortalObjectSyncMessage";
        ObjectTypeEnum[ObjectTypeEnum["FileManagerFile"] = 'FILE_MANAGER_FILE'] = "FileManagerFile";
        ObjectTypeEnum[ObjectTypeEnum["FileManagerFolder"] = 'FILE_MANAGER_FOLDER'] = "FileManagerFolder";
        ObjectTypeEnum[ObjectTypeEnum["SequenceStepEnrollment"] = 'SEQUENCE_STEP_ENROLLMENT'] = "SequenceStepEnrollment";
        ObjectTypeEnum[ObjectTypeEnum["Approval"] = 'APPROVAL'] = "Approval";
        ObjectTypeEnum[ObjectTypeEnum["ApprovalStep"] = 'APPROVAL_STEP'] = "ApprovalStep";
        ObjectTypeEnum[ObjectTypeEnum["CtaVariant"] = 'CTA_VARIANT'] = "CtaVariant";
        ObjectTypeEnum[ObjectTypeEnum["Unknown"] = 'UNKNOWN'] = "Unknown";
    })(ObjectTypeEnum = PublicImportError.ObjectTypeEnum || (PublicImportError.ObjectTypeEnum = {}));
})(PublicImportError = exports.PublicImportError || (exports.PublicImportError = {}));
