// @hubspot/api-client/lib/codegen/automation/actions/model/actionFunctionIdentifier.js
"use strict";
/**
 * Custom Workflow Actions
 * Create custom workflow actions
 *
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionFunctionIdentifier = void 0;
/**
* A serverless function associated with this custom workflow action.
*/
class ActionFunctionIdentifier {
    static getAttributeTypeMap() {
        return ActionFunctionIdentifier.attributeTypeMap;
    }
}
exports.ActionFunctionIdentifier = ActionFunctionIdentifier;
ActionFunctionIdentifier.discriminator = undefined;
ActionFunctionIdentifier.attributeTypeMap = [
    {
        "name": "functionType",
        "baseName": "functionType",
        "type": "ActionFunctionIdentifier.FunctionTypeEnum"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
(function (ActionFunctionIdentifier) {
    let FunctionTypeEnum;
    (function (FunctionTypeEnum) {
        FunctionTypeEnum[FunctionTypeEnum["PreActionExecution"] = 'PRE_ACTION_EXECUTION'] = "PreActionExecution";
        FunctionTypeEnum[FunctionTypeEnum["PreFetchOptions"] = 'PRE_FETCH_OPTIONS'] = "PreFetchOptions";
        FunctionTypeEnum[FunctionTypeEnum["PostFetchOptions"] = 'POST_FETCH_OPTIONS'] = "PostFetchOptions";
    })(FunctionTypeEnum = ActionFunctionIdentifier.FunctionTypeEnum || (ActionFunctionIdentifier.FunctionTypeEnum = {}));
})(ActionFunctionIdentifier = exports.ActionFunctionIdentifier || (exports.ActionFunctionIdentifier = {}));
