// @hubspot/api-client/lib/codegen/automation/actions/model/actionFunction.js
"use strict";
/**
 * Custom Workflow Actions
 * Create custom workflow actions
 *
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionFunction = void 0;
/**
* A serverless function associated with this custom workflow action.
*/
class ActionFunction {
    static getAttributeTypeMap() {
        return ActionFunction.attributeTypeMap;
    }
}
exports.ActionFunction = ActionFunction;
ActionFunction.discriminator = undefined;
ActionFunction.attributeTypeMap = [
    {
        "name": "functionSource",
        "baseName": "functionSource",
        "type": "string"
    },
    {
        "name": "functionType",
        "baseName": "functionType",
        "type": "ActionFunction.FunctionTypeEnum"
    },
    {
        "name": "id",
        "baseName": "id",
        "type": "string"
    }
];
(function (ActionFunction) {
    let FunctionTypeEnum;
    (function (FunctionTypeEnum) {
        FunctionTypeEnum[FunctionTypeEnum["PreActionExecution"] = 'PRE_ACTION_EXECUTION'] = "PreActionExecution";
        FunctionTypeEnum[FunctionTypeEnum["PreFetchOptions"] = 'PRE_FETCH_OPTIONS'] = "PreFetchOptions";
        FunctionTypeEnum[FunctionTypeEnum["PostFetchOptions"] = 'POST_FETCH_OPTIONS'] = "PostFetchOptions";
    })(FunctionTypeEnum = ActionFunction.FunctionTypeEnum || (ActionFunction.FunctionTypeEnum = {}));
})(ActionFunction = exports.ActionFunction || (exports.ActionFunction = {}));
