// @hubspot/api-client/lib/codegen/automation/actions/model/fieldTypeDefinition.js
"use strict";
/**
 * Custom Workflow Actions
 * Create custom workflow actions
 *
 * The version of the OpenAPI document: v4
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FieldTypeDefinition = void 0;
/**
* The data type expected by an input field.
*/
class FieldTypeDefinition {
    static getAttributeTypeMap() {
        return FieldTypeDefinition.attributeTypeMap;
    }
}
exports.FieldTypeDefinition = FieldTypeDefinition;
FieldTypeDefinition.discriminator = undefined;
FieldTypeDefinition.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "type",
        "baseName": "type",
        "type": "FieldTypeDefinition.TypeEnum"
    },
    {
        "name": "fieldType",
        "baseName": "fieldType",
        "type": "FieldTypeDefinition.FieldTypeEnum"
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<Option>"
    },
    {
        "name": "optionsUrl",
        "baseName": "optionsUrl",
        "type": "string"
    },
    {
        "name": "referencedObjectType",
        "baseName": "referencedObjectType",
        "type": "FieldTypeDefinition.ReferencedObjectTypeEnum"
    }
];
(function (FieldTypeDefinition) {
    let TypeEnum;
    (function (TypeEnum) {
        TypeEnum[TypeEnum["String"] = 'string'] = "String";
        TypeEnum[TypeEnum["Number"] = 'number'] = "Number";
        TypeEnum[TypeEnum["Bool"] = 'bool'] = "Bool";
        TypeEnum[TypeEnum["Datetime"] = 'datetime'] = "Datetime";
        TypeEnum[TypeEnum["Enumeration"] = 'enumeration'] = "Enumeration";
        TypeEnum[TypeEnum["Date"] = 'date'] = "Date";
        TypeEnum[TypeEnum["PhoneNumber"] = 'phone_number'] = "PhoneNumber";
        TypeEnum[TypeEnum["CurrencyNumber"] = 'currency_number'] = "CurrencyNumber";
        TypeEnum[TypeEnum["Json"] = 'json'] = "Json";
        TypeEnum[TypeEnum["ObjectCoordinates"] = 'object_coordinates'] = "ObjectCoordinates";
    })(TypeEnum = FieldTypeDefinition.TypeEnum || (FieldTypeDefinition.TypeEnum = {}));
    let FieldTypeEnum;
    (function (FieldTypeEnum) {
        FieldTypeEnum[FieldTypeEnum["Booleancheckbox"] = 'booleancheckbox'] = "Booleancheckbox";
        FieldTypeEnum[FieldTypeEnum["Checkbox"] = 'checkbox'] = "Checkbox";
        FieldTypeEnum[FieldTypeEnum["Date"] = 'date'] = "Date";
        FieldTypeEnum[FieldTypeEnum["File"] = 'file'] = "File";
        FieldTypeEnum[FieldTypeEnum["Number"] = 'number'] = "Number";
        FieldTypeEnum[FieldTypeEnum["Phonenumber"] = 'phonenumber'] = "Phonenumber";
        FieldTypeEnum[FieldTypeEnum["Radio"] = 'radio'] = "Radio";
        FieldTypeEnum[FieldTypeEnum["Select"] = 'select'] = "Select";
        FieldTypeEnum[FieldTypeEnum["Text"] = 'text'] = "Text";
        FieldTypeEnum[FieldTypeEnum["Textarea"] = 'textarea'] = "Textarea";
        FieldTypeEnum[FieldTypeEnum["CalculationEquation"] = 'calculation_equation'] = "CalculationEquation";
        FieldTypeEnum[FieldTypeEnum["CalculationRollup"] = 'calculation_rollup'] = "CalculationRollup";
        FieldTypeEnum[FieldTypeEnum["CalculationScore"] = 'calculation_score'] = "CalculationScore";
        FieldTypeEnum[FieldTypeEnum["CalculationReadTime"] = 'calculation_read_time'] = "CalculationReadTime";
        FieldTypeEnum[FieldTypeEnum["Unknown"] = 'unknown'] = "Unknown";
    })(FieldTypeEnum = FieldTypeDefinition.FieldTypeEnum || (FieldTypeDefinition.FieldTypeEnum = {}));
    let ReferencedObjectTypeEnum;
    (function (ReferencedObjectTypeEnum) {
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Contact"] = 'CONTACT'] = "Contact";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Company"] = 'COMPANY'] = "Company";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Deal"] = 'DEAL'] = "Deal";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Engagement"] = 'ENGAGEMENT'] = "Engagement";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Ticket"] = 'TICKET'] = "Ticket";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Owner"] = 'OWNER'] = "Owner";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Product"] = 'PRODUCT'] = "Product";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["LineItem"] = 'LINE_ITEM'] = "LineItem";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["BetDeliverableService"] = 'BET_DELIVERABLE_SERVICE'] = "BetDeliverableService";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Content"] = 'CONTENT'] = "Content";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Conversation"] = 'CONVERSATION'] = "Conversation";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["BetAlert"] = 'BET_ALERT'] = "BetAlert";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Portal"] = 'PORTAL'] = "Portal";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Quote"] = 'QUOTE'] = "Quote";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["FormSubmissionInbounddb"] = 'FORM_SUBMISSION_INBOUNDDB'] = "FormSubmissionInbounddb";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Quota"] = 'QUOTA'] = "Quota";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Unsubscribe"] = 'UNSUBSCRIBE'] = "Unsubscribe";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Communication"] = 'COMMUNICATION'] = "Communication";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["FeedbackSubmission"] = 'FEEDBACK_SUBMISSION'] = "FeedbackSubmission";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Attribution"] = 'ATTRIBUTION'] = "Attribution";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SalesforceSyncError"] = 'SALESFORCE_SYNC_ERROR'] = "SalesforceSyncError";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["RestorableCrmObject"] = 'RESTORABLE_CRM_OBJECT'] = "RestorableCrmObject";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Hub"] = 'HUB'] = "Hub";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["LandingPage"] = 'LANDING_PAGE'] = "LandingPage";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ProductOrFolder"] = 'PRODUCT_OR_FOLDER'] = "ProductOrFolder";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Task"] = 'TASK'] = "Task";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Form"] = 'FORM'] = "Form";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["MarketingEmail"] = 'MARKETING_EMAIL'] = "MarketingEmail";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["AdAccount"] = 'AD_ACCOUNT'] = "AdAccount";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["AdCampaign"] = 'AD_CAMPAIGN'] = "AdCampaign";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["AdGroup"] = 'AD_GROUP'] = "AdGroup";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Ad"] = 'AD'] = "Ad";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Keyword"] = 'KEYWORD'] = "Keyword";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Campaign"] = 'CAMPAIGN'] = "Campaign";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SocialChannel"] = 'SOCIAL_CHANNEL'] = "SocialChannel";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SocialPost"] = 'SOCIAL_POST'] = "SocialPost";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SitePage"] = 'SITE_PAGE'] = "SitePage";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["BlogPost"] = 'BLOG_POST'] = "BlogPost";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Import"] = 'IMPORT'] = "Import";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Export"] = 'EXPORT'] = "Export";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Cta"] = 'CTA'] = "Cta";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["TaskTemplate"] = 'TASK_TEMPLATE'] = "TaskTemplate";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["AutomationPlatformFlow"] = 'AUTOMATION_PLATFORM_FLOW'] = "AutomationPlatformFlow";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ObjectList"] = 'OBJECT_LIST'] = "ObjectList";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Note"] = 'NOTE'] = "Note";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["MeetingEvent"] = 'MEETING_EVENT'] = "MeetingEvent";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Call"] = 'CALL'] = "Call";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Email"] = 'EMAIL'] = "Email";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["PublishingTask"] = 'PUBLISHING_TASK'] = "PublishingTask";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ConversationSession"] = 'CONVERSATION_SESSION'] = "ConversationSession";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ContactCreateAttribution"] = 'CONTACT_CREATE_ATTRIBUTION'] = "ContactCreateAttribution";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Invoice"] = 'INVOICE'] = "Invoice";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["MarketingEvent"] = 'MARKETING_EVENT'] = "MarketingEvent";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["ConversationInbox"] = 'CONVERSATION_INBOX'] = "ConversationInbox";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Chatflow"] = 'CHATFLOW'] = "Chatflow";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["MediaBridge"] = 'MEDIA_BRIDGE'] = "MediaBridge";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Sequence"] = 'SEQUENCE'] = "Sequence";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["SequenceStep"] = 'SEQUENCE_STEP'] = "SequenceStep";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Forecast"] = 'FORECAST'] = "Forecast";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Snippet"] = 'SNIPPET'] = "Snippet";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Template"] = 'TEMPLATE'] = "Template";
        ReferencedObjectTypeEnum[ReferencedObjectTypeEnum["Unknown"] = 'UNKNOWN'] = "Unknown";
    })(ReferencedObjectTypeEnum = FieldTypeDefinition.ReferencedObjectTypeEnum || (FieldTypeDefinition.ReferencedObjectTypeEnum = {}));
})(FieldTypeDefinition = exports.FieldTypeDefinition || (exports.FieldTypeDefinition = {}));
