// @hubspot/api-client/lib/codegen/crm/extensions/cards/model/cardDisplayProperty.js
"use strict";
/**
 * CRM cards
 * Allows an app to extend the CRM UI by surfacing custom cards in the sidebar of record pages. These cards are defined up-front as part of app configuration, then populated by external data fetch requests when the record page is accessed by a user.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardDisplayProperty = void 0;
/**
* Definition for a card display property.
*/
class CardDisplayProperty {
    static getAttributeTypeMap() {
        return CardDisplayProperty.attributeTypeMap;
    }
}
exports.CardDisplayProperty = CardDisplayProperty;
CardDisplayProperty.discriminator = undefined;
CardDisplayProperty.attributeTypeMap = [
    {
        "name": "name",
        "baseName": "name",
        "type": "string"
    },
    {
        "name": "label",
        "baseName": "label",
        "type": "string"
    },
    {
        "name": "dataType",
        "baseName": "dataType",
        "type": "CardDisplayProperty.DataTypeEnum"
    },
    {
        "name": "options",
        "baseName": "options",
        "type": "Array<DisplayOption>"
    }
];
(function (CardDisplayProperty) {
    let DataTypeEnum;
    (function (DataTypeEnum) {
        DataTypeEnum[DataTypeEnum["Boolean"] = 'BOOLEAN'] = "Boolean";
        DataTypeEnum[DataTypeEnum["Currency"] = 'CURRENCY'] = "Currency";
        DataTypeEnum[DataTypeEnum["Date"] = 'DATE'] = "Date";
        DataTypeEnum[DataTypeEnum["Datetime"] = 'DATETIME'] = "Datetime";
        DataTypeEnum[DataTypeEnum["Email"] = 'EMAIL'] = "Email";
        DataTypeEnum[DataTypeEnum["Link"] = 'LINK'] = "Link";
        DataTypeEnum[DataTypeEnum["Numeric"] = 'NUMERIC'] = "Numeric";
        DataTypeEnum[DataTypeEnum["String"] = 'STRING'] = "String";
        DataTypeEnum[DataTypeEnum["Status"] = 'STATUS'] = "Status";
    })(DataTypeEnum = CardDisplayProperty.DataTypeEnum || (CardDisplayProperty.DataTypeEnum = {}));
})(CardDisplayProperty = exports.CardDisplayProperty || (exports.CardDisplayProperty = {}));
