// @rockset/client/dist/tests/errors.test.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
var index_1 = require("../index");
var basePath = process.env.ROCKSET_APISERVER;
var apikey = process.env.ROCKSET_APIKEY;
if (!apikey) {
    throw 'No ROCKSET_APIKEY specified. Please specify an environment variable ROCKSET_APIKEY with your Rockset key. eg: $ export ROCKSET_APIKEY=...';
}
if (!basePath) {
    throw 'No ROCKSET_APISERVER specified. Please specify an environment variable ROCKSET_APISERVER with your Rockset key. eg: $ export ROCKSET_APISERVER=...';
}
console.log('Specified apiserver to hit:', basePath);
var rockset = index_1.default(apikey, basePath);
afterAll(function () { });
describe('error handling', function () {
    var _this = this;
    test('properly parses query error model', function () { return __awaiter(_this, void 0, void 0, function () {
        var out;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, rockset.queries
                        .query({
                        sql: {
                            // Syntax error - typo in COUNT
                            query: 'SELECT COUT(*) FROM _events;',
                        },
                    })
                        .catch(function (e) {
                        return expect(e).toMatchObject({
                            line: 1,
                            column: 8,
                            error_id: 'function_not_found',
                            message: 'Function `COUT` not found.',
                            type: 'QueryError',
                        });
                    })];
                case 1:
                    out = _a.sent();
                    expect(out).toBeUndefined();
                    return [2 /*return*/];
            }
        });
    }); });
    test('properly parses non-JSON error responses', function () { return __awaiter(_this, void 0, void 0, function () {
        var rocksetBadGateway, out;
        return __generator(this, function (_a) {
            switch (_a.label) {
                case 0:
                    rocksetBadGateway = index_1.default(apikey, 'https://httpstat.us/502?path=');
                    return [4 /*yield*/, rocksetBadGateway.queries
                            .query({
                            sql: {
                                query: 'SELECT 5',
                            },
                        })
                            .catch(function (e) {
                            return expect(e).toMatchObject({
                                code: 502,
                                message: 'Bad Gateway',
                            });
                        })];
                case 1:
                    out = _a.sent();
                    expect(out).toBeUndefined();
                    return [2 /*return*/];
            }
        });
    }); });
});
