// @hubspot/api-client/lib/codegen/cms/hubdb/model/batchResponseHubDbTableRowV3WithErrors.js
"use strict";
/**
 * HubDB endpoints
 * HubDB is a relational data store that presents data as rows, columns, and cells in a table, much like a spreadsheet. HubDB tables can be added or modified [in the HubSpot CMS](https://knowledge.hubspot.com/cos-general/how-to-edit-hubdb-tables), but you can also use the API endpoints documented here. For more information on HubDB tables and using their data on a HubSpot site, see the [CMS developers site](https://designers.hubspot.com/docs/tools/hubdb). You can also see the [documentation for dynamic pages](https://designers.hubspot.com/docs/tutorials/how-to-build-dynamic-pages-with-hubdb) for more details about the `useForPages` field.  HubDB tables support `draft` and `published` versions. This allows you to update data in the table, either for testing or to allow for a manual approval process, without affecting any live pages using the existing data. Draft data can be reviewed, and published by a user working in HubSpot or published via the API. Draft data can also be discarded, allowing users to go back to the published version of the data without disrupting it. If a table is set to be `allowed for public access`, you can access the published version of the table and rows without any authentication by specifying the portal id via the query parameter `portalId`.
 *
 * The version of the OpenAPI document: v3
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchResponseHubDbTableRowV3WithErrors = void 0;
class BatchResponseHubDbTableRowV3WithErrors {
    static getAttributeTypeMap() {
        return BatchResponseHubDbTableRowV3WithErrors.attributeTypeMap;
    }
}
exports.BatchResponseHubDbTableRowV3WithErrors = BatchResponseHubDbTableRowV3WithErrors;
BatchResponseHubDbTableRowV3WithErrors.discriminator = undefined;
BatchResponseHubDbTableRowV3WithErrors.attributeTypeMap = [
    {
        "name": "status",
        "baseName": "status",
        "type": "BatchResponseHubDbTableRowV3WithErrors.StatusEnum"
    },
    {
        "name": "results",
        "baseName": "results",
        "type": "Array<HubDbTableRowV3>"
    },
    {
        "name": "numErrors",
        "baseName": "numErrors",
        "type": "number"
    },
    {
        "name": "errors",
        "baseName": "errors",
        "type": "Array<StandardError>"
    },
    {
        "name": "requestedAt",
        "baseName": "requestedAt",
        "type": "Date"
    },
    {
        "name": "startedAt",
        "baseName": "startedAt",
        "type": "Date"
    },
    {
        "name": "completedAt",
        "baseName": "completedAt",
        "type": "Date"
    },
    {
        "name": "links",
        "baseName": "links",
        "type": "{ [key: string]: string; }"
    }
];
(function (BatchResponseHubDbTableRowV3WithErrors) {
    let StatusEnum;
    (function (StatusEnum) {
        StatusEnum[StatusEnum["Pending"] = 'PENDING'] = "Pending";
        StatusEnum[StatusEnum["Processing"] = 'PROCESSING'] = "Processing";
        StatusEnum[StatusEnum["Canceled"] = 'CANCELED'] = "Canceled";
        StatusEnum[StatusEnum["Complete"] = 'COMPLETE'] = "Complete";
    })(StatusEnum = BatchResponseHubDbTableRowV3WithErrors.StatusEnum || (BatchResponseHubDbTableRowV3WithErrors.StatusEnum = {}));
})(BatchResponseHubDbTableRowV3WithErrors = exports.BatchResponseHubDbTableRowV3WithErrors || (exports.BatchResponseHubDbTableRowV3WithErrors = {}));
